Source/WebCore/platform/graphics/cocoa/GraphicsContextGLCocoa.mm

```diff
@@ -964,6 +964,29 @@ static bool needsEAGLOnMac()
     func(*displayImage);
 }
 
+void GraphicsContextGLCocoa::insertFinishedSignalOrInvoke(Function<void()> signal)
+{
+    if (!contextAttributes().useMetal) {
+        signal();
```

Comment:
Is this the correct thing to do? It feels like `Function<void()> signal` is acting as a promise and the documentation says that signal is called with the rendering is finished, but in this case it isn't. Should the `signal` take a bool indicating whether rendering is finished, or not?

Reply:
the docs also say "If not possible, calls the `signal`."

E.g. there's nothing to be done if we don't have a primitive to do that thing being asked to do.

The intention is that the code is structured so that the caller obtains the previous behavior, e.g scheduled rendering, with the same code structure. This would be the logic the caller would anyway default to, if they knew "backend doesn't support singling on finished."